File: apps/mobile/lib/services/data_export_service.dart
import '../core/network/api_client.dart';
import '../core/storage/secure_store.dart';

class DataExport {
  DataExport({
    required this.id,
    required this.status,
    required this.format,
    required this.payload,
    required this.createdAt,
    required this.completedAt,
  });

  final String id;
  final String status;
  final String format;
  final Map<String, dynamic> payload;
  final DateTime? createdAt;
  final DateTime? completedAt;

  factory DataExport.fromJson(Map<String, dynamic> json) {
    return DataExport(
      id: json['id']?.toString() ?? '',
      status: json['status']?.toString() ?? '',
      format: json['format']?.toString() ?? '',
      payload: json['payload'] is Map<String, dynamic>
          ? json['payload'] as Map<String, dynamic>
          : <String, dynamic>{},
      createdAt:
          DateTime.tryParse(json['createdAt']?.toString() ?? ''),
      completedAt:
          DateTime.tryParse(json['completedAt']?.toString() ?? ''),
    );
  }
}

class DataExportService {
  DataExportService({SecureStore? store})
      : _client = ApiClient(store ?? SecureStore());

  final ApiClient _client;

  Future<DataExport?> fetchLatest() async {
    final data = await _client.get('/users/me/data-export', auth: true);
    if (data is! Map<String, dynamic>) return null;
    final export = data['export'];
    if (export is! Map<String, dynamic>) return null;
    return DataExport.fromJson(export);
  }

  Future<DataExport?> requestExport() async {
    final data = await _client.post('/users/me/data-export', auth: true);
    if (data is! Map<String, dynamic>) return null;
    final export = data['export'];
    if (export is! Map<String, dynamic>) return null;
    return DataExport.fromJson(export);
  }
}

File: apps/web/src/services/data-export-service.ts
import { apiClient } from '../adapters/api-client';

export interface DataExport {
  id: string;
  status: string;
  format: string;
  payload: Record<string, unknown>;
  createdAt?: string | null;
  completedAt?: string | null;
}

class DataExportService {
  async fetchLatest(): Promise<DataExport | null> {
    const data = await apiClient.get<{ export?: DataExport | null }>(
      '/users/me/data-export',
      { auth: true }
    );
    return data.export ?? null;
  }

  async requestExport(): Promise<DataExport | null> {
    const data = await apiClient.post<{ export?: DataExport | null }>(
      '/users/me/data-export',
      { auth: true }
    );
    return data.export ?? null;
  }
}

export const dataExportService = new DataExportService();
